plugin/plugin.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 the Drone Authors. All rights reserved.
+// Use of this source code is governed by the Blue Oak Model License
+// that can be found in the LICENSE file.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// Args provides plugin execution arguments.
+type Args struct {
+	Pipeline
+
+	// Level defines the plugin log level.
+	Level string `envconfig:"PLUGIN_LOG_LEVEL"`
+
+	ApiKey string `envconfig:"PLUGIN_NUGET_APIKEY"`
+	NugetUri string `envconfig:"PLUGIN_NUGET_URI"`
+	PackageLocation string `envconfig:"PLUGIN_PACKAGE_LOCATION"`
+}
+
+const globalNugetUri = "https://api.nuget.org/v3/index.json"
+
+func fileExists(name string) bool {
+	if _, err := os.Stat(name); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
+
+func validateAndSetArgs(args *Args) error {
+	if args.ApiKey == ""{
+		return fmt.Errorf ("nuget api key must be set in settings")
+	}
+	if args.NugetUri == "" {
+		args.NugetUri = globalNugetUri
+	}
+	if args.PackageLocation != "" && !fileExists(args.PackageLocation){
+		return fmt.Errorf ("the package location: %s does not exist", args.PackageLocation)
```

Comment:
might be better to have 2 different errors, one for the setting being blank, and one for the file not existing.
otherwise if its blank, the error will be 
`"the package location:  does not exist"`